### PR TITLE
Remove data_dir pointer from TabletMeta

### PR DESCRIPTION
**Pull request by chaoyli**

### CODE REVIEW

File: be/src/olap/tablet_manager.cpp [hunk 1]
```diff
@@ -686,7 +686,7 @@ TabletSharedPtr TabletManager::find_best_tablet_to_compaction(CompactionType com
 
 OLAPStatus TabletManager::load_tablet_from_meta(DataDir* data_dir, TTabletId tablet_id,
         TSchemaHash schema_hash, const std::string& meta_binary) {
-    TabletMetaSharedPtr tablet_meta(new TabletMeta());
+    TabletMetaSharedPtr tablet_meta(new (nothrow) TabletMeta());
```
Comment (kangpinghuang) [inline be/src/olap/tablet_manager.cpp]:
```suggestion
    TabletMetaSharedPtr tablet_meta(new(nothrow) TabletMeta());
```

File: be/src/olap/snapshot_manager.cpp [hunk 2]
```diff
@@ -497,7 +497,7 @@ OLAPStatus SnapshotManager::_append_single_delta(
         const TSnapshotRequest& request, DataDir* store) {
     OLAPStatus res = OLAP_SUCCESS;
     string root_path = store->path();
-    TabletMetaSharedPtr new_tablet_meta(new(nothrow) TabletMeta(store));
+    TabletMetaSharedPtr new_tablet_meta(new (nothrow) TabletMeta());
```
Comment (kangpinghuang) [inline be/src/olap/snapshot_manager.cpp]:
```suggestion
    TabletMetaSharedPtr new_tablet_meta(new(nothrow) TabletMeta());
```

**Review by kangpinghuang — APPROVED** [review]

File: be/src/olap/snapshot_manager.cpp (hunk 2 above)
Comment (chaoyli) [reply be/src/olap/snapshot_manager.cpp]: all new (nothrow) has space

File: be/src/olap/tablet_manager.cpp (hunk 1 above)
Comment (chaoyli) [reply be/src/olap/tablet_manager.cpp]: all new (nothrow) has space

**Pull request merged**